src/CodeGen_RS_Dev.cpp

```diff
@@ -0,0 +1,552 @@
+        user_assert(b_y->value.type().width == 1)
+            << "Image_load/store y coordinate is not scalar.\n";
+        args.push_back(sym_get(b_name->value.as<StringImm>()->value));
+        args.push_back(codegen(b_x->value));
+        args.push_back(codegen(b_y->value));
+    } else {
+        // Use all three coordinates to retrieve single byte.
+        user_assert(b_name == NULL && b_x == NULL && b_y == NULL &&
+                    ramp_c == NULL);
+        args.push_back(sym_get(name.as<StringImm>()->value));
+        args.push_back(codegen(x));
+        args.push_back(codegen(y));
+        args.push_back(codegen(c));
+    }
+    return args;
+}
+
+void CodeGen_RS_Dev::visit(const Call *op) {
+    if (op->call_type == Call::Intrinsic) {
+        if (op->name == Call::image_load || op->name == Call::image_store) {
+            //
+            // image_load(<image name>, <buffer>, <x>, <x-extent>, <y>,
+            // <y-extent>, <c>, <c-extent>)
+            // or
+            // image_store(<image name>, <buffer>, <x>, <y>, <c>, <value>)
+            //
+            const int index_name = 0;
+            const int index_x = op->name == Call::image_load ? 2 : 2;
+            const int index_y = op->name == Call::image_load ? 4 : 3;
+            const int index_c = op->name == Call::image_load ? 6 : 4;
+            vector<Value *> args =
+                add_x_y_c_args(op->args[index_name], op->args[index_x],
+                               op->args[index_y], op->args[index_c]);
+
+            if (op->name == Call::image_store) {
+                args.insert(args.begin() + 1, codegen(op->args[5]));
+            }
+
+            debug(2) << "Generating " << op->type.width
+                     << "byte-wide call with " << args.size() << " args:\n";
+            if (debug::debug_level >= 2) {
+                int i = 1;
+                for (Value *arg : args) {
+                    debug(2) << " #" << i++ << ":";
+                    arg->getType()->dump();
+                    arg->dump();
+                }
+            }
+
+            llvm::Function *func = op->name == Call::image_load
+                                       ? fetch_GetElement_func(op->type.width)
+                                       : fetch_SetElement_func(op->type.width);
+            value = builder->CreateCall(func, args);
+            return;
+        }
+    }
+    CodeGen_LLVM::visit(op);
+}
+
+string CodeGen_RS_Dev::march() const { return "armv7"; }
+
+string CodeGen_RS_Dev::mcpu() const { return "none"; }
+
+string CodeGen_RS_Dev::mattrs() const { return "linux-gnueabi"; }
+
+bool CodeGen_RS_Dev::use_soft_float_abi() const {
+    // Taken from CodeGen_ARM::use_soft_float_abit.
+    return target.bits == 32;
+}
+
+llvm::Triple CodeGen_RS_Dev::get_target_triple() const {
+    return Triple(Triple::normalize(march() + "-" + mcpu() + "-" + mattrs()));
+}
+
+struct AndroidBitcodeWrapper {
+    uint32_t Magic;
```

Comment:
The data structure and write function are coming from https://android.googlesource.com/platform/frameworks/compile/libbcc/+/master/include/bcinfo/BitcodeWrapper.h, so I believe worth preserving.
Adding comment regarding this.